obsreport/obsreport_processor.go

```diff
@@ -54,134 +84,228 @@ type ProcessorSettings struct {
 
 // NewProcessor creates a new Processor.
 func NewProcessor(cfg ProcessorSettings) (*Processor, error) {
-	return &Processor{
-		level:    cfg.ProcessorCreateSettings.MetricsLevel,
-		mutators: []tag.Mutator{tag.Upsert(obsmetrics.TagKeyProcessor, cfg.ProcessorID.String(), tag.WithTTL(tag.TTLNoPropagation))},
-	}, nil
+	return newProcessor(cfg, featuregate.GetRegistry())
+}
+
+// Deprecated: [v0.65.0] use NewProcessor.
```

Comment:
This PR will probably land after we release 0.66.0 (happening now), so, this needs to be updated.

Reply:
Thanks for pointing this out! Removed this deprecated function now.